ts/src/gate.ts

```diff
@@ -6263,6 +6264,45 @@ export default class gate extends Exchange {
         return await this.privateFuturesPostSettleDualMode (this.extend (request, query));
     }
 
+    async fetchUnderlyingAssets (marketType: string = undefined, params = {}) {
+        /**
+         * @method
+         * @name gate#fetchUnderlyingAssets
+         * @description fetches the market ids of underlying assets for a specific contract market type
+         * @see https://www.gate.io/docs/developers/apiv4/en/#list-all-underlyings
+         * @param {string|undefined} marketType the contract market type, 'option', 'swap' or 'future', the default is 'option'
```

Comment:
@Dan-krm the correct way of representing optional parameters is by doing ` * @param {string} [type] ` (using the square brackets around it)